watcher/watcher.go

```diff
@@ -0,0 +1,156 @@
+package watcher
+
+import (
+	"log"
+	"os"
+	"path/filepath"
+	"strings"
+	"time"
+
+	"github.com/Tonkpils/snag/exchange"
+	fsn "gopkg.in/fsnotify.v1"
+)
+
+const rebuild = "rebuild"
+
+type Watcher interface {
+	Watch(string) error
+}
+
+type FSWatcher struct {
+	ex           exchange.SendListener
+	fsn          *fsn.Watcher
+	done         chan struct{}
+	mtimes       map[string]time.Time
+	watching     map[string]struct{}
+	watchDir     string
+	ignoredItems []string
+}
+
+func New(ex exchange.SendListener, ignoredItems []string) (Watcher, error) {
```

Comment:
Ok, I think I understand better now where you're coming form. I can see how that would work for the watcher since it only ever sends one event with no data. However, how would that work for the builder? 

The builder sends 3 (potentially a 4th even for logs) events with data (in this case the name). Would it get a `chan myStruct` and then parse it accordingly? What you mind making a commit to show how this would look?

---

What I like about the components using the exchange is the consistency in the type. We could also move the registering of events into the components themselves. For instance the builder could do something like

```go
ex.Listen("rebuild", b.Build)
```

inside it's constructor which would alleviate the main class from any sort of orchestration (although i'm not sure if this would make it *cleaner*).